src/Microsoft.Extensions.Logging.AzureWebApps/Internal/WebAppsLogConfigurationReader.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions.Logging.AzureWebApps.Internal
+{
+    /// <summary>
+    /// Represents the default implementation of the <see cref="IWebAppsLogConfigurationReader"/>
+    /// </summary>
+    public class WebAppsLogConfigurationReader : IWebAppsLogConfigurationReader
+    {
+        private readonly IConfigurationRoot _configuration;
+        private readonly string _fileLogFolder;
+
+        private WebAppsLogConfiguration _latestConfiguration;
+        private IDisposable _changeSubscription;
+
+        /// <inheritdoc />
+        public event EventHandler<WebAppsLogConfiguration> OnConfigurationChanged;
+
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="WebAppsLogConfigurationReader"/> class
+        /// </summary>
+        /// <param name="context">The context in which the reader runs</param>
+        public WebAppsLogConfigurationReader(IWebAppsContext context)
+        {
+            if (!context.IsRunningInAzureWebApp)
+            {
+                _latestConfiguration = WebAppsLogConfiguration.Disabled;
+            }
+            else
+            {
+                _fileLogFolder = Path.Combine(context.HomeFolder, @"LogFiles\Application");
+                var settingsFile = Path.Combine(context.HomeFolder, @"Diagnostics\settings.json");
+
+                _configuration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddJsonFile(settingsFile, optional: true, reloadOnChange: true)
```

Comment:
Not in this case. We don't want an environment variable to be able to overwrite the config file.